metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1917,4 +1964,19 @@ class MetalsLanguageServer(
     })
   }
 
+  private object JsonString {
+    def unapply(json: Any): Option[String] = json match {
+      case p: JsonPrimitive if p.isString() => Some(p.getAsString())
```

Comment:
Why you have: `p: JsonPrimitive if p.isString()` and below you can match on JsonString?

Reply:
when I match on JsonString below, I use this exactly method `unapply`.